File: src/main/java/io/dustin/finstage/user/infra/mapper/UserMapper.java
package io.dustin.finstage.user.infra.mapper;

import io.dustin.finstage.user.domain.User;
import io.dustin.finstage.user.infra.entity.UserEntity;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class UserMapper {

    public static User toDomain(UserEntity entity) {
        return User.fromEntity(
                entity.getId(),
                entity.getEmail(),
                entity.getPassword()
        );
    }

    public static UserEntity toEntity(User domain) {
        return new UserEntity(
                domain.getId(),
                domain.getEmail(),
                domain.getPassword()
        );
    }
}

File: src/main/kotlin/io/dustin/finstage/user/infra/mapper/UserMapper.kt
package io.dustin.finstage.user.infra.mapper

import io.dustin.finstage.user.domain.User
import io.dustin.finstage.user.infra.entity.UserEntity

object UserMapper {

    fun toDomain(entity: UserEntity): User? {
        return entity.id?.let {
            User.fromEntity(
                id = it,
                email = entity.email,
                password = entity.password
        )
        }
    }

    fun toEntity(domain: User): UserEntity {
        return UserEntity(
                id = domain.id,
                email = domain.email,
                password = domain.password
        )
    }
}
